### PR TITLE
add list-template in svc layer

### LINKED ISSUE #265: [Feature Request] Kusion provides API for integration tools
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now the extension is accomplishing its goals by executing the kusion binary for each time and then resolve the `stdout & stderr`, but since the stdout/stderr of kusion binary are not designed for tools integration, the output messages like this are disturbing:
image

the vscode-kusion extension requires kusion to provide APIs for tools to integrate kusion's capacities more elegantly and normatively. 

**Describe the feature you'd like:**

At least a few Go APIs of kusion are required:
- compile
- preview
- apply
- watch
- list projects
- list stacks
- list templates

### ListTemplates API

- API id：`init.ListTemplates`
- params:

| name | type | required | default value | description | example |
| ----- | ----- | -------- | ------------- | ----------- | --------- |
| online | boolean | optional | false | determines if kusion downloads templates from online repo | `true`, `false` |
| url | string | optional | `"https://github.com/KusionStack/kusion-templates"` | the online repo url, meaningful only when `online==true`|  |
| fsPath | string | optional |  | the fsPath to find the templates, meaningful only when `online == false` |`/path/to/templates` |

- returns:

| name | type | description | example |
| ----- | ----- | ----------- | -------- |
| templateList | Template[] | list of templates found | |
| err | Error | error code and message | `{"code":"invalidPath"， "message":"path not exists (or repo url not exists)"}` |

```go
type Template struct {
    name string
    description string
    content string // now the template could be the content of `kusion.yaml` file under the template directory: for example https://github.com/KusionStack/kusion-templates/blob/main/code-city/kusion.yaml
}

type Error struct {
    code  string
    message string
}
```

### PR DESCRIPTION
**Pull request by healthjyk**
#### What type of PR is this?
feature

#### What this PR does / why we need it:
start of construction of svc layer, provide list-template layer 

#### Which issue(s) this PR fixes:

Fixes #265

#### Special notes for your reviewer:
Please check the input and output for func ListTemplates

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by healthjyk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by amyXia1994** [comment]
recheck

**Comment by healthjyk** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pkg/svc/template.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+func ListTemplates(o *ListTemplateOptions) ([]*Template, error) {
+	// Validate listTemplateOptions
+	if err := o.Validate(); err != nil {
+		return nil, WrapInvalidArgumentErr(err)
+	}
+
+	// retrieve the template repo.
+	var templateNamePathOrURL string
+	if o.Online {
+		templateNamePathOrURL = o.URL
```
Comment (SparkYuan) [inline pkg/svc/template.go:56]: The name of this filed`templateNamePathOrURL` is confusing. You could create a new PR to make this field more clear.

**Pull request closed without merging**